.github/workflows/docs.yml

```diff
@@ -0,0 +1,24 @@
+---
+name: s2n-tls Documentation
+
+on:
+  pull_request:
```

Comment:
Should this be on merge?  We don't want new docs until  a PR is approved ?

Reply:
IMO a PR that introduces a warning to the API docs should be blocked. I'm thinking we could add a separate job to update the `gh-pages` branch on merge or we could just push docs as a release artifact